broker-core/src/main/java/io/zeebe/broker/workflow/state/WorkflowPersistenceCache.java

```diff
@@ -0,0 +1,305 @@
+/*
+ * Zeebe Broker Core
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.zeebe.broker.workflow.state;
+
+import io.zeebe.broker.workflow.deployment.data.DeploymentRecord;
+import io.zeebe.broker.workflow.deployment.data.DeploymentResource;
+import io.zeebe.broker.workflow.deployment.data.Workflow;
+import io.zeebe.broker.workflow.model.ExecutableWorkflow;
+import io.zeebe.broker.workflow.model.transformation.BpmnTransformer;
+import io.zeebe.logstreams.state.StateController;
+import io.zeebe.model.bpmn.Bpmn;
+import io.zeebe.model.bpmn.BpmnModelInstance;
+import io.zeebe.util.buffer.BufferUtil;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.agrona.DirectBuffer;
+import org.agrona.ExpandableArrayBuffer;
+import org.agrona.collections.Int2ObjectHashMap;
+import org.agrona.collections.Long2ObjectHashMap;
+import org.agrona.collections.LongHashSet;
+import org.agrona.io.DirectBufferInputStream;
+import org.rocksdb.ColumnFamilyHandle;
+
+public class WorkflowPersistenceCache {
+  private static final byte[] WORKFLOWS_FAMILY_NAME = "workflows".getBytes();
+  private static final byte[] WORKFLOWS_BY_ID_AND_VERSION_FAMILY_NAME =
+      "workflowsByIdAndVersion".getBytes();
+
+  private final BpmnTransformer transformer = new BpmnTransformer();
+
+  private final Map<DirectBuffer, Int2ObjectHashMap<DeployedWorkflow>>
+      workflowsByProcessIdAndVersion = new HashMap<>();
+  private final Map<DirectBuffer, DeployedWorkflow> latestWorkflowsByProcessId = new HashMap<>();
+
+  private final StateController rocksDbWrapper;
+  private final ColumnFamilyHandle workflowsHandle;
+  private final ColumnFamilyHandle workflowsByIdAndVersionHandle;
```

Comment:
This contains two types of mappings:

(ID + VERSION) => workflow
(ID) => workflow (for latest versions)

There is a chance that the (ID) key for one workflow is equal to the (ID + VERSION) key of another, and then they will overwrite their entries in the table. I think its safer to have a separate column family for latest versions.